File: LeetCode/Easy/Valid Palindrome/Go/validPalindrome.go
func isPalindrome(s string) bool {
	n := len(s)
	filter := ""
	for i := 0; i < n; i++ {
		if (s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z') {
			filter += strings.ToLower(string(s[i]))
		}
		if s[i] >= '0' && s[i] <= '9' {
			filter += string(s[i])
		}
	}

	//reverse the string
	reverse_string := ""
	n = len(filter)
	for i := n - 1; i >= 0; i-- {
		reverse_string += string(filter[i])
	}
	return filter == reverse_string
}

File: LeetCode/Easy/Valid Palindrome/Java/validPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        StringBuilder sb = new StringBuilder();
        for (int i=0; i<s.length(); i++)
            if (Character.isAlphabetic(s.charAt(i)) || Character.isDigit(s.charAt(i))) 
                sb.append(s.charAt(i));
                
        return sb.toString().equals(sb.reverse().toString());
    }
}
